src/main/kotlin/org/kamiblue/client/module/modules/combat/CombatSetting.kt

```diff
@@ -161,27 +161,27 @@ internal object CombatSetting : Module(
         if (CrystalAura.isDisabled && CrystalBasePlace.isDisabled && CrystalESP.isDisabled && player.ticksExisted % 4 != 0) return
 
         val eyePos = player.getPositionEyes(1f) ?: Vec3d.ZERO
-        val cacheList = ArrayList<Pair<BlockPos, Triple<Float, Float, Double>>>()
+        val cacheList = ArrayList<Pair<BlockPos, CombatManager.CombatCalculation>>()
         val target = CombatManager.target
         val prediction = target?.let { getPrediction(it) }
 
         for (pos in getPlacePos(target, player, 8f)) {
             val dist = eyePos.distanceTo(pos.toVec3dCenter(0.0, 0.5, 0.0))
             val damage = target?.let { calcCrystalDamage(pos, it, prediction?.first, prediction?.second) } ?: 0.0f
             val selfDamage = calcCrystalDamage(pos, player)
-            cacheList.add(Pair(pos, Triple(damage, selfDamage, dist)))
+            cacheList.add(Pair(pos, CombatManager.CombatCalculation(damage, selfDamage, dist)))
         }
 
-        CombatManager.placeMap = LinkedHashMap<BlockPos, Triple<Float, Float, Double>>(cacheList.size).apply {
-            putAll(cacheList.sortedByDescending { triple -> triple.second.first })
```

Comment:
sortedWith already takes Comparator as parameter

Reply:
sortedWith takes a Comparator yes, but it's not compatible with the CombatCalculations comparator which is why it's 

compareBy(comparator) { it.second } which unwraps the key part of the pair to allow for the comparator usage.